boards/samd10-xmini/Makefile.dep

```diff
@@ -0,0 +1,10 @@
+ifneq (,$(filter saul_default,$(USEMODULE)))
+  USEMODULE += saul_gpio
+endif
+
+# Use Picolibc to reduce ROM usage
+FEATURES_REQUIRED += picolibc
+USEMODULE += picolibc
+
+# limit the stack size so most tests and examples compile
+CFLAGS += -DTHREAD_STACKSIZE_DEFAULT=512
```

Comment:
I have a suspicion that this is the reason for the failed tests.

Reply:
How about you drop this for now? That way we don't get heart attacks during the next release testing due to the failing tests. And users of SAMD10 boards can still hand-tune their stacks to get tests running.

Also, I think that overwriting this via `CFLAGS` from within RIOT is generally not a good idea. If a user does the same and the user's `CFLAGS` end up being overwritten by this, great confusion would result.

Reply:
So I suppose we should also drop this from `stk3200` and `nucleo-l011k4`